cmd/kops/validate_cluster.go

```diff
@@ -64,10 +64,15 @@ func NewCmdValidateCluster(f *util.Factory, out io.Writer) *cobra.Command {
 		Long:    validateLong,
 		Example: validateExample,
 		Run: func(cmd *cobra.Command, args []string) {
-			err := RunValidateCluster(f, cmd, args, os.Stdout, options)
+			result, err := RunValidateCluster(f, cmd, args, os.Stdout, options)
 			if err != nil {
 				exitWithError(err)
 			}
+			// We want the validate command to exit non-zero if validation found a problem,
+			// even if we didn't really hit an error during validation.
+			if len(result.PodFailures) != 0 {
```

Comment:
We obviously care about more than PodFailures, but this will be fixed later once the series is merged